Core/MIPS/IR/IRCompBranch.cpp

```diff
@@ -40,9 +41,8 @@
 #define _SA MIPS_GET_SA(op)
 #define _POS  ((op>> 6) & 0x1F)
 #define _SIZE ((op>>11) & 0x1F)
-#define _IMM16 (signed short)(op & 0xFFFF)
 #define _IMM26 (op & 0x03FFFFFF)
-#define TARGET16 ((int)((uint32_t)(int)_IMM16 << 2))
+#define TARGET16 ((int)(SignExtend16To32(op) << 2))
```

Comment:
Just for clarity I'd write this as op & 0xFFFF because it looks kinda weird to sign extend a value that's 32-bit already. should be optimized out.

Reply:
Or actually .. It's not that bad since it's clear that it goes from 16 bits anyway .. Hm. Maybe leave it as is.

Reply:
Yeah, I intentionally removed the masking in all these cases with the intent of that being clear, but maybe it's too far.  I'm on the fence.

-[Unknown]